### DISCUSSION TITLE
issue: default value and validate function unexpected behaviour

### DISCUSSION THREAD

**Discussion by ilteoood** [react-hook-form/react-hook-form#10911] [Q&A]
### Version Number

7.46.1

### Codesandbox/Expo snack

https://codesandbox.io/s/crimson-shape-vlxyhc?file=/src/App.tsx

### Steps to reproduce

Issue 1:
- Open the repro
- see that the value defined in the reset function is not reflected to the input
- type something
- see that the validate function is never invoked

Issue 2
- Open the repro
- move the content of the reset function to the `defaultValue` parameter
- remove the useEffect hook
- see that the value defined in the reset function is not reflected to the input
- type something
- see that the validate function in this case is invoked

### Expected behaviour

1) The default value should always be applied to the input
2) The validate function should be called in both cases

P.S. unfortunately I can't use the `useFieldArray` hook, because the component which is rendering my array items allows me to define just the react component to render the single item, so I don't have access to the `map` loop to look at the field id.

### What browsers are you seeing the problem on?

Firefox, Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
I don't think we support such register name `invalid register name: `{...register("0.firstName", {`

**Reply by ilteoood** [reply]
I looked at it from the example provided for the useFieldArray.
Even if I wrap my array within another field, just as the example, the behavior seems the same
